docs/configuration.md

```diff
@@ -403,6 +403,13 @@ support overriding).
 
 ```
 
+:::{warning}
+
+Setting defines through `cmake.args` in `pyproject.toml` is discouraged because
+this cannot be later altered via command line.
```

Comment:
```suggestion
this cannot be later altered via command line. Use `cmake.define` instead.
```

Reply:
We could also flip the order of this, possibly? But yeah, `cmake.args` is meant for args, not defines.

Reply:
I think the current order is preferable so that `args` can be passed transparently from other tools without worrying about the inverse issue.